### PR TITLE
Adaptive tract loading

### PR DESCRIPTION
**Pull request by coconut750750**
- Frontend renders tracts that are currently in the window. This makes the initial loading of the map quicker.

ezgif-2-96d687bd64b6

### CODE REVIEW

File: frontend/src/resources/stateLayers.js [hunk 1]
```diff
@@ -1,118 +1,69 @@
...
+  { sourceLayer: "west_virginia-8jz02z", sourceURL: "mapbox://meghabyte.4lueicdj", id: "54" },
+  { sourceLayer: "wisconsin-32yyxh", sourceURL: "mapbox://meghabyte.3iwzwwxv", id: "55" },
+  { sourceLayer: "wyoming-758cbr", sourceURL: "mapbox://meghabyte.2zqpgoup", id: "56" },
 ];
 
 export let sourceURLs = [];
 
 stateLayers.map(stateLayer => {
   sourceURLs.push(stateLayer.sourceURL);
 });
```
Comment (caiyingshen) [inline frontend/src/resources/stateLayers.js]: do we still need this at all after your change?

File: frontend/src/components/Home.js [hunk 2]
```diff
@@ -12,6 +12,7 @@ import Graph from "./Graph.js";
 mapboxgl.accessToken =
   "pk.eyJ1IjoibWVnaGFieXRlIiwiYSI6ImNrMXlzbDYxNzA3NXYzbnBjbWg5MHd2bGgifQ._sJyE87zG6o5k32efYbrAA";
 
+const MIN_TRACT_ZOOM = 6;
```
Comment (caiyingshen) [inline frontend/src/components/Home.js]: whats the rationale behind changing the zoom?

File: frontend/src/components/Home.js [hunk 3]
```diff
@@ -72,67 +74,34 @@ class Home extends React.Component {
...
     });
 
+    this.map.on("moveend", () => {
+      const zoom = this.map.getZoom().toFixed(2);
+      if (zoom > MIN_TRACT_ZOOM) {
+        let tractIDs = this.getRenderedTracts();
+        console.log('rendered tracts:', tractIDs.length);
+        if (tractIDs.length > 0) {
+          this.updateRenderedTracts(tractIDs);
+        }
```
Comment (caiyingshen) [inline frontend/src/components/Home.js]: if someone is in zoom level 10 but then zooms out to level 4, do the tracts just stay on the screen without anything else loading? 

also pls remove console.logs heh

File: frontend/src/resources/stateLayers.js (hunk 1 above)
Comment (coconut750750) [reply frontend/src/resources/stateLayers.js]: prob not

File: frontend/src/components/Home.js (hunk 3 above)
Comment (coconut750750) [reply frontend/src/components/Home.js]: ya some tracts stay on

File: frontend/src/components/Home.js (hunk 2 above)
Comment (coconut750750) [reply frontend/src/components/Home.js]: changed it back lol

**Review by chyku — APPROVED** [review]
looks gucci

**Review by caiyingshen — APPROVED** [review]
beep beep 

ty for having a big brain

**Pull request merged**